pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -89,25 +92,13 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		r:   r,
 		app: app,
 	}
+	if app.Status.LatestRevision != nil {
+		// record previous app revision name
+		handler.previousRevisionName = app.Status.LatestRevision.Name
+	}
```

Comment:
what's used for?

Reply:
If app revision is changed, app controller will update app's status before applying resources. So we should record this previous revision at the beginning of reconciliation.

Reply:
I just found it used in test, does it really necessary？

Reply:
it's very important for GC and used in apply.go.